arch/arm/core/cortex_m/fpu.c

```diff
@@ -16,7 +16,7 @@
 
 void z_arm_save_fp_context(struct fpu_ctx_full *buffer)
 {
-#if defined(CONFIG_FPU_SHARING)
+#if defined(CONFIG_FPU)
```

Comment:
From the commit message:

> z_arm_save_fp_context and z_arm_restore_fp_context
save and restore fpu context regardless of the
CONFIG_FPU_SHARING setting.

Please add the rationale for this in the commit message.

Reply:
> From the commit message:
> 
> > z_arm_save_fp_context and z_arm_restore_fp_context
> > save and restore fpu context regardless of the
> > CONFIG_FPU_SHARING setting.
> 
> Please add the rationale for this in the commit message.

Done.